lint/src/main/java/permissions/dispatcher/detectors/PsiCallNeedsPermissionDetector.java

```diff
@@ -0,0 +1,117 @@
+package permissions.dispatcher.detectors;
+
+import com.android.tools.lint.detector.api.Detector;
+import com.android.tools.lint.detector.api.Issue;
+import com.android.tools.lint.detector.api.JavaContext;
+import com.android.tools.lint.detector.api.Scope;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiReferenceExpression;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class PsiCallNeedsPermissionDetector extends Detector implements Detector.JavaPsiScanner {
```

Comment:
presumably current implementation doesn't pass `UastCallNeedsPermissionDetectorTest`, I found there's a logic mistake on https://github.com/permissions-dispatcher/PermissionsDispatcher/pull/368 so we need to align the logic of `PsiCallNeedsPermissionDetector` to `UastCallNeedsPermissionDetector` 😅

Reply:
Interesting! Let's migrate the test to #368's version, then compare the results again. :)